### PR TITLE
Add bugfix from bugs_and_glitches.md to BUGFIX

### LINKED ISSUE #1215: Item flicker fix in bugs_and_glitches.md is outdated and incomplete
[This fix](https://github.com/pret/pokeemerald/blob/master/docs/bugs_and_glitches.md#scrolling-through-items-in-the-bag-causes-the-image-to-flicker) contains a reference to an undocumented field that has since been documented, and is also dependent on this fix which should be referenced:
DiscordCanary_qBjVwDBDHX

Without including the above, the doc fix as is produces the following result:
broken fix

### LINKED ISSUE #1514: Bag Item Sprite Flickering fix in bugs_and_glitches.md does not work
Since this piece of documentation is a part of the main repository, I figured I should open an issue ticket for it.

**[The fix](https://github.com/pret/pokeemerald/blob/master/docs/bugs_and_glitches.md#scrolling-through-items-in-the-bag-causes-the-image-to-flicker)**, on top of being slightly out of date as the `spriteId` variable of the `BagMenu` struct was renamed to `spritesId`, does not work like it should even on a clean copy of Pokeemerald without any other modifications.

%pn_1327

### PR DESCRIPTION
**Pull request by coltongit**
Item flicker fix now in `BUGFIX`
Supersedes and resolves #1680
Fixes #1215
Fixes #1514

## Description

Moved the bugfix described in bugs_and_glitches.md to `BUGFIX`

### CODE REVIEW

File: src/item_menu_icons.c [hunk 1]
```diff
@@ -544,7 +544,20 @@ void AddBagItemIconSprite(u16 itemId, u8 id)
...
+        gSprites[spriteId[id ^ 1]].invisible = TRUE;
+
+    if (spriteId[id] != SPRITE_NONE)
+    {
+        DestroySpriteAndFreeResources(&gSprites[spriteId[id]]);
+        spriteId[id] = SPRITE_NONE;
+    }
+    #else
+        RemoveBagSprite(id + ITEMMENUSPRITE_ITEM);
+    #endif
```
Comment (Sneed69) [inline src/item_menu_icons.c:560]:
```suggestion
#ifdef BUGFIX
    u8 *spriteId = &gBagMenu->spriteIds[ITEMMENUSPRITE_ITEM];

if (spriteId[id ^ 1] != SPRITE_NONE)
    gSprites[spriteId[id ^ 1]].invisible = TRUE;

if (spriteId[id] != SPRITE_NONE)
{
    DestroySpriteAndFreeResources(&gSprites[spriteId[id]]);
    spriteId[id] = SPRITE_NONE;
}
#else
    RemoveBagSprite(id + ITEMMENUSPRITE_ITEM);
#endif
```

Comment (LOuroboros) [reply src/item_menu_icons.c:560]: I was gonna talk about the syntax too, but then I noticed it's already inconsistent.

Check `src/daycare.c` and `src/battle_script_commands.c` for example.

**Comment by GriffinRichards** [comment]
Non-obvious bug fixes like this should have a comment describing the issue they're resolving

**Comment by coltongit** [comment]
> Non-obvious bug fixes like this should have a comment describing the issue they're resolving

Fixed in latest commit

**Pull request merged**